### PR TITLE
[cli] Support pnpm in `getUpdateCommand()`

### PR DESCRIPTION
**Pull request by TooTallNate**
We're currently showing bad instructions when CLI was installed via pnpm:

image

This adds support for it by leveraging our existing `scanParentDirs()` logic from build-utils.

### CODE REVIEW

**Comment by cb1kenobi** [comment]
FWIW, this may solve the one scenario, but it doesn't solve mine local machine where `process.argv[1]` equals `"/usr/loca/bin/vercel"`.
image
Interestingly, I'm not sure why it's selecting `yarn` over `npm`. The code before suggested `npm`:
image

**Comment by TooTallNate** [comment]
@cb1kenobi Try again with latest.

**Comment by cb1kenobi** [comment]
Still no global flag, but now it's suggesting `pnpm`!
image

**Review by cb1kenobi — DISMISSED** [review]

**Comment by cb1kenobi** [comment]
Looks good! Tested npm, pnpm, yarn v1 + nvm on macOS as well as npm and yarn v1 + nvm on Windows.

File: packages/cli/src/util/get-update-command.ts [hunk 1]
```diff
@@ -100,13 +82,24 @@ async function isGlobal() {
...
+  let { cliType, lockfilePath } = await scanParentDirs(
+    dirname(dirname(entrypoint))
+  );
+  if (!lockfilePath) {
+    // Global installs for npm do not have a lockfile
+    cliType = 'npm';
+  }
+  const yarn = cliType === 'yarn';
+
+  let install = yarn ? 'add' : 'i';
```
Comment (styfle) [inline packages/cli/src/util/get-update-command.ts:95]: I think `pnpm add` should be used too, right?

https://pnpm.io/cli/add

**Review by cb1kenobi — APPROVED** [review]

File: packages/cli/src/util/get-update-command.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/cli/src/util/get-update-command.ts:95]: Both work, but ya from the docs it looks like `add` would be more appropriate for this use-case.

**Pull request merged**